compdb/README.md

```diff
@@ -0,0 +1,26 @@
+JSON Compilation Database
+=========================
+
+Contains a small program to create a JSON compilation database file for the current project
+which can be read by Clang / libclang etc.
+
+See https://clang.llvm.org/docs/JSONCompilationDatabase.html for more information about
+the format.
+
+You may find it convenient to add an alias in your project for it:
+```
+alias ["clangcompdb"]
+cmd = run ///cc//compdb
+desc = Generate Clang compile_commands.json file
+```
+
+Limitations
+-----------
+
+At present it does not build any dependencies so the some completions may not
+work if they have dynamically generated inputs etc (similarly they will not
```

Comment:
What do you mean by dynamically generated? Build rules that are inputs to another rule, or some sort of post build action?

Reply:
The former, yeah. I think this tool would be agnostic to post-build actions.

Reply:
Although actually you're probably right, it wouldn't know about any actions created by a post-build function.

I'm not gonna worry about that right now until we know more about whether this is useful / relevant; the only real way to solve is to do a whole-repo build which I'd at least want to be a flag since it's expensive and solving a relatively rare case.